### PR TITLE
[QnA Maker] Active learning enabled flag added, More test case added,…

### PR DESCRIPTION
**Pull request by trojenguri**
Active learning enabled flag added, 
More test case added for QnA Maker action
Bug fixes on QnA Maker dialog
Feature link: https://github.com/microsoft/botbuilder-dotnet/issues/2595

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerActionBuilder.cs [hunk 1]
```diff
@@ -140,27 +140,27 @@ private async Task<DialogTurnResult> CallGenerateAnswerAsync(WaterfallStepContex
...
             // Take this value from GetAnswerResponse 
-            var isActiveLearningEnabled = true;
+            var isActiveLearningEnabled = response.ActiveLearningEnabled;
 
-            stepContext.Values[QnAData] = new List<QueryResult>(response);
+            stepContext.Values[QnAData] = new List<QueryResult>(response.Answers);
             
             // Check if active learning is enabled.
-            if (isActiveLearningEnabled)
+            if (isActiveLearningEnabled && response.Answers.Any() && response.Answers.First().Score <= 0.95)
```
Comment (carlosscastro) [inline libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerActionBuilder.cs]: Should this be configurable?
Magic numbers are always suspicious. Is this a widely accepted QnA parameter?

**Comment by carlosscastro** [comment]
REquested changes only to get more info on where the 0.95 comes from.

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerActionBuilder.cs (hunk 1 above)
Comment (trojenguri) [reply libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerActionBuilder.cs]: No, It is not configurable. Its upper cap for calling active learning, 
We check if the score is below 95%, then only we consider feedback for that code, else when confidence score is very high (ie above 95%), we don't check for feedbacks.

Comment (trojenguri) [reply libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerActionBuilder.cs]: I have modified this const to a private variable.

**Pull request merged**